packages/editor/src/components/widgets/code-editor/useCodeEditor.ts

```diff
@@ -15,29 +15,34 @@ export type ModifyAction = (value: string) => string;
 export const useMonacoEditor = (options?: { modifyAction?: ModifyAction; scriptAreaDatatypeEditor?: boolean }) => {
   const [editor, setEditor] = useState<monaco.editor.IStandaloneCodeEditor>();
 
-  const modifyEditor = (value: string, type: BrowserType) => {
+  const modifyEditor = (value: BrowserValue) => {
     if (!editor) {
       return;
     }
     const selection = editor.getSelection();
     if (!selection) {
       return;
     }
-
-    if (type === 'type' && options?.scriptAreaDatatypeEditor) {
-      const textWithLineBreak = 'import ' + value + ';\n';
-      const parts = value.split('.');
-      const shortValue = parts.length > 1 ? parts.pop() : value;
+    if (value.firstLineValue) {
+      let cursorValue = value.cursorValue;
+      if (options?.scriptAreaDatatypeEditor) {
+        const parts = cursorValue.split('.');
+        cursorValue = parts.length > 1 ? String(parts.pop()) : cursorValue;
+      }
```

Comment:
Theoretically, I could also perform this check in the TypeBrowser and completely remove the flag scriptAreaDatatypeEditor. However, in the TypeBrowser, I would do this using location.includes('code'). You can find this check already in the TypeBrowser, as I used it there to ensure that "import" is only created in the code block. However, I'm not sure if using .includes('code') covers all cases or allows for too many cases. @ivy-lli What do you think?
Also I removed the BrowserType because I currently don't need it for any condition.